# Critical Values

#### Critical Values
- It is common, to interpret the results of statistical hypothesis tests using a p-value.
- Not all implementations of statistical tests return p-values for interpreting the outcome of the test..
- In these cases, an alternatives method must be employed for interpreting the calculated test statistic directly, such as critical values.
- A statistic calculated by a statistical hypothesis test can be interpreted using critical values from the distribution of the test statistic.
- In addition, critical values are used when estimating the expected intervals for observations from a population, such as in tolerance intervals.

#### Interpreting Critical Values
- Some tests do not return a p-value. Instead, they might return a test statistic value (such as mean) from a specific data distribution that can be interpreted in the context of critical values.
- A critical value is a value from the distribution of the test statistic after which point the result is significant and the null hypothesis can be rejected.

Test Statistic < Critical Value : NOT significant result, fail to reject null hypothesis (H0).

Test Statistic ≥ Critical Value : Significant result, reject null hypothesis (H1).


- It requires that the distribution of the test statistic be known and how to sample the distribution to retrieve the critical value. The p-value is calculated from the critical value. 

Again, the meaning of the result is similar in that the chosen significance level is a probabilistic decision on the rejection or failure of rejection of the base assumption of the test given the data. Results are presented in the same way as with a p-value, as either significance level or confidence level. For example, if a normality test was calculated and the test statistic was compared to the critical value at the 5% significance level, results could be stated as:

The test found that the data sample was normal, failing to reject the null hypothesis at a 5% significance level.

Or:

The test found that the data was normal, failing to reject the null hypothesis at a 95% confidence level.

Some examples of statistical hypothesis tests and their distributions from which critical values can be calculated are as follows:
- #### Z Test: Gaussian distribution.
- #### Student’s t-Test: Student’s t-distribution.
- #### Chi-Squared Test: Chi-Squared distribution.
- #### F-distribution.

Critical values are also used when defining intervals for expected (or unexpected) observations in distributions. Calculating and using critical values may be appropriate when quantifying the uncertainty of estimated statistics or intervals such as confidence intervals and tolerance intervals. Note, a p-value can be calculated from a test statistic by retrieving the probability from the test statistics cumulative density function (CDF).

#### What Is a Critical Value
- A critical value is defined in the context of the population distribution and a probability.
- An observation from the population with a value equal to or lesser than a critical value with the given probability.
- Can express this mathematically as follows:

Pr[X ≤ critical value] = probability

- Where Pr is the calculation of probability.
- X are observations from the population.
- Critical value is the calculated critical value.
- Probability is the chosen probability.

- Critical values are calculated using a mathematical function where the probability is provided as an argument.
- For most common distributions, the value cannot be calculated analytically; instead it must be estimated using numerical methods.
- Historically it is common for tables of pre-calculated critical values to be provided in the appendices of statistics textbooks for reference purposes.
- Critical values are used in statistical significance testing.
- The probability is often expressed as a significance, denoted as the lowercase Greek letter alpha (α), which is the inverted probability.

probability = 1−alpha

Standard alpha values are used when calculating critical values, chosen for historical reasons and continually used for consistency reasons.

These alpha values include:
- 1% (alpha=0.01)
- 5% (alpha=0.05)
- 10% (alpha=0.10)

Critical values provide an alternative and equivalent way to interpret statistical hypothesis tests to the p-value.


## Terms

### Quantile Function
In probability and statistics, the **quantile function**, associated with a probability distribution of a random variable, specifies the value of the random variable such that the probability of the variable being less than or equal to that value equals the given probability. It is also called the **percent-point function** or **inverse cumulative distribution function**.

### Percentage Point
A **percentage point** or **percent point** is the unit for the arithmetic difference of two percentages. For example, moving up from 40% to 44% is a 4 percentage point increase, but is an actual 10 percent increase in what is being measured.  percentages indicate ratios, not differences.

- The percent point function can be calculated in SciPy using the ppf() function on a given distribution.
- It should also be noted that you can also calculate the ppf() using the inverse survival function called isf() in SciPy.
- These alternate approaches may be used in third party code.

## --------------------------------------------------------------------------------

### gaussian percent point function

The example below calculates the percent point function for 95% on the standard Gaussian distribution.

In [1]:
from scipy.stats import norm

# define probability
p = 0.95

# retrieve value <= probability
value = norm.ppf(p)
print(value)

# confirm with cdf
p = norm.cdf(value)
print(p)

1.6448536269514722
0.95


#### Sample output from calculating critical values for the Gaussian distribution.
- The above example first prints the value that marks 95% or less of the observations from the distribution of about 1.65.
- This value is then confirmed by retrieving the probability of the observation from the CDF, which returns 95%, as expected.
- We can see that the value 1.65 aligns with our expectation with regard to the number of standard deviations from the mean that cover 95% of the distribution in the 68-95-99.7 rule.


##### Calculate critical values for 90% for the Gaussian distributions.

In [2]:
# define probability
p = 0.90

# retrieve value <= probability
value = norm.ppf(p)
print(value)

# confirm with cdf
p = norm.cdf(value)
print(p)

1.2815515655446004
0.8999999999999999


##### Calculate critical values for 99% for the Gaussian distributions.

In [3]:
# define probability
p = 0.99

# retrieve value <= probability
value = norm.ppf(p)
print(value)

# confirm with cdf
p = norm.cdf(value)
print(p)

2.3263478740408408
0.99


## --------------------------------------------------------------------------------

### student t-distribution percent point function

The example below calculates the percentage point function for 95% on the standard Student’s t-distribution with 10 degrees of freedom.

In [4]:
from scipy.stats import t

# define probability
p = 0.95
df = 10

# retrieve value <= probability
value = t.ppf(p, df)
print(value)

# confirm with cdf
p = t.cdf(value, df)
print(p)

1.8124611228107335
0.949999999999923


#### Sample output from calculating critical values for the t distribution.¶
- The above example returns the value of about 1.812 or less that covers 95% of the observations from the chosen distribution.
- The probability of the value is then confirmed (with minor rounding error) via the CDF.

##### Calculate critical values for 90% for the t distribution.

In [5]:
# define probability
p = 0.90
df = 10

# retrieve value <= probability
value = t.ppf(p, df)
print(value)

# confirm with cdf
p = t.cdf(value, df)
print(p)

1.3721836411102863
0.8999999999999926


##### Calculate critical values for 99% for the t distribution.

In [6]:
# define probability
p = 0.99
df = 10

# retrieve value <= probability
value = t.ppf(p, df)
print(value)

# confirm with cdf
p = t.cdf(value, df)
print(p)

2.763769457447889
0.9899999999885913


## --------------------------------------------------------------------------------

### chi-squared percent point function

The example below calculates the percentage point function for 95% on the standard Chi-Squared distribution with 10 degrees of freedom.

In [7]:
from scipy.stats import chi2

# define probability
p = 0.95
df = 10

# retrieve value <= probability
value = chi2.ppf(p, df)
print(value)

# confirm with cdf
p = chi2.cdf(value, df)
print(p)

18.307038053275143
0.95


#### Sample output from calculating critical values for the Chi-Squared distribution.

#### Sample output from calculating critical values for the Chi-Squared distribution.
- The above example first calculates the value of 18.3 or less that covers 95% of the observations from the distribution.
- The probability of this observation is confirmed by using it as input to the CDF.

##### Calculate critical values for 90% for the Chi-Squared distribution.

In [8]:
from scipy.stats import chi2

# define probability
p = 0.90
df = 10

# retrieve value <= probability
value = chi2.ppf(p, df)
print(value)

# confirm with cdf
p = chi2.cdf(value, df)
print(p)

15.987179172105263
0.9


##### Calculate critical values for 99% for the Chi-Squared distribution.

In [9]:
from scipy.stats import chi2

# define probability
p = 0.99
df = 10

# retrieve value <= probability
value = chi2.ppf(p, df)
print(value)

# confirm with cdf
p = chi2.cdf(value, df)
print(p)

23.209251158954356
0.99
